sdk/adp/adp-test-rest/src/aDPRestClient.ts

```diff
@@ -0,0 +1,40 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+import { getClient, ClientOptions } from "@azure-rest/core-client";
+import { TokenCredential } from "@azure/core-auth";
+import { ADPRestClient } from "./clientDefinitions";
+
+/**
+ * Initialize a new instance of the class ADPRestClient class.
+ * @param credentials type: TokenCredential
+ */
+export default function createClient(
+  credentials: TokenCredential,
+  options: ClientOptions = {}
+): ADPRestClient {
+  const baseUrl = options.baseUrl ?? "undefined";
```

Comment:
Cna you figure out why this is undefined ?

Reply:
Yes I identified the root cause and issue tracked here https://github.com/Azure/autorest.typescript/issues/1654